nc_test4/tst_strings3.c

```diff
@@ -0,0 +1,53 @@
+/* This is part of the netCDF package. Copyright 2005 University
+   Corporation for Atmospheric Research/Unidata See COPYRIGHT file for
+   conditions of use. See www.unidata.ucar.edu for more info.
+
+   Test netcdf-4 character type.
+
+   $Id$
+*/
+
+#include <config.h>
+#include <nc_tests.h>
+
+#define FILE_NAME "tst_strings3.nc"
+#define ATT_NAME "WC"
+
+int
+main(int argc, char **argv)
+{
+   printf("\n*** Testing netcdf-4 character type.\n");
+   printf("*** testing a string attribute with non ASCII characters...");
+   {
+#define ATT_LEN 1
+      size_t att_len;
+      int ndims, nvars, natts, unlimdimid;
+      nc_type att_type;
+      int ncid, i;
+      char *data_in[ATT_LEN];
+      char *data[ATT_LEN] = {"abcdeABCDEF12345\x00\xAA\xBB\xCC\xDD\xEE\xFF"};
+
+      if (nc_create(FILE_NAME, NC_NETCDF4, &ncid)) ERR;
+      if (nc_put_att(ncid, NC_GLOBAL, ATT_NAME, NC_CHAR, ATT_LEN, data)) ERR;
+      if (nc_inq(ncid, &ndims, &nvars, &natts, &unlimdimid)) ERR;
+      if (ndims != 0 || nvars != 0 || natts != 1 || unlimdimid != -1) ERR;
+      if (nc_inq_att(ncid, NC_GLOBAL, ATT_NAME, &att_type, &att_len)) ERR;
+      if (att_type != NC_CHAR || att_len != ATT_LEN) ERR;
+      if (nc_close(ncid)) ERR;
+
+      /* Check it out. */
+      if (nc_open(FILE_NAME, NC_NOWRITE, &ncid)) ERR;
+      if (nc_inq(ncid, &ndims, &nvars, &natts, &unlimdimid)) ERR;
+      if (ndims != 0 || nvars != 0 || natts != 1 || unlimdimid != -1) ERR;
+      if (nc_inq_att(ncid, NC_GLOBAL, ATT_NAME, &att_type, &att_len)) ERR;
+      if (att_type != NC_CHAR || att_len != ATT_LEN) ERR;
+      if (nc_get_att(ncid, NC_GLOBAL, ATT_NAME, data_in)) ERR;
+      for (i = 0; i < att_len; i++)
+         if (strcmp(data_in[i], data[i])) ERR;
```

Comment:
The test failed on this line. Is there an easy way to get the error message to print the strings that failed the comparison?